src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -46,80 +77,1175 @@ const ActorInit En_Kakasi_InitVars = {
+        this->unk244.x = ((float)D_8097203C.x) + this->unk22C.x; // cast req
+        this->unk244.y = ((float)D_8097203C.y) + this->unk22C.y;
+        this->unk244.z = ((float)D_8097203C.z) + this->unk22C.z;
+        Math_Vec3f_Copy(&this->unk214, &this->unk238);
+        Math_Vec3f_Copy(&this->unk220, &this->unk244);
+        func_8096FAAC(this, globalCtx);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_8096FE00.s")
+    if (this->unkCounter1A4 < 0xF) {
+        this->unkCounter1A4++;
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80970008.s")
+    this->actor.shape.rot.y = this->actor.shape.rot.y + 0x3000;
+    Math_SmoothStepToS(&this->unk190, 500, 5, 50, 0);
+    if ((globalCtx->gameplayFrames & 3) == 0) {
+        Math_Vec3f_Copy(&tempWorldPos, &this->actor.world.pos);
+        tempWorldPos.y = this->actor.floorHeight;
+        tempWorldPos.x += randPlusMinusPoint5Scaled(2.0f);
+        tempWorldPos.z += randPlusMinusPoint5Scaled(2.0f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_8097006C.s")
+        if (globalCtx->sceneNum == SCENE_8ITEMSHOP) {
+            EffectSsGSplash_Spawn(globalCtx, &tempWorldPos, 0, 0, 0, randPlusMinusPoint5Scaled(100.0f) + 200.0f);
+            func_800F0568(globalCtx, &tempWorldPos, 0x32, NA_SE_EV_BOMB_DROP_WATER);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_809705E4.s")
+        } else {
+            func_800BBDAC(globalCtx, &this->actor, &this->actor.world.pos, this->actor.shape.shadowScale - 20.0f, 5,
+                          4.0f, 200, 10, 1);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_AKINDONUTS_HIDE);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80970658.s")
+    Math_ApproachF(&this->actor.shape.yOffset, -6000.0f, 0.5f, 200.0f);
+    if (fabsf(this->actor.shape.yOffset + 6000.0f) < 10.0f) {
+        gSaveContext.weekEventReg[79] |= 8;
+        func_800B7298(globalCtx, &this->actor, 6);
+        ActorCutscene_Stop(this->actorCutscenes[0]);
+        this->aboveGroundStatus = ENKAKASI_ABOVE_GROUND_TYPE;
+        this->songSummonDist = 80.0f;
+        EnKakasi_SetupIdleUnderground(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80970740.s")
+void EnKakasi_SetupIdleUnderground(EnKakasi* this) {
+    this->actor.shape.yOffset = -7000.0;
+    this->actor.draw = NULL;
+    this->actor.flags |= 0x8000000;
+    this->unkState196 = 5;
+    this->actionFunc = EnKakasi_IdleUnderground;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80970978.s")
+void EnKakasi_IdleUnderground(EnKakasi* this, GlobalContext* globalCtx) {
+    if ((gSaveContext.weekEventReg[79] & 0x8) && this->actor.xzDistToPlayer < this->songSummonDist &&
+        (gGameInfo->data[2401] != 0 || globalCtx->msgCtx.unk1202A == 0xD)) {
```

Comment:
We have `tools/regconvert.py` too!
```suggestion
        (BREG(1) != 0 || globalCtx->msgCtx.unk1202A == 0xD)) {
```